src/Microsoft.Azure.Functions.Worker.Extensions.Mcp/Configuration/InputSchemaGenerator.cs

```diff
@@ -0,0 +1,223 @@
+                continue;
+            }
+
+            if (TryProcessToolPropertyParameter(parameter, properties, required))
+            {
+                continue;
+            }
+
+            if (TryProcessPocoParameter(parameter, properties, required))
+            {
+                continue;
+            }
+        }
+
+        return CreateSchemaNode(properties, required);
+    }
+
+    /// <summary>
+    /// Determines if a parameter should be included in schema generation.
+    /// </summary>
+    private static bool ShouldIncludeParameter(ParameterInfo parameter)
+    {
+        // Skip ToolInvocationContext parameters
+        if (parameter.ParameterType == typeof(ToolInvocationContext))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Processes parameters with McpToolPropertyAttribute.
+    /// </summary>
+    private static bool TryProcessToolPropertyParameter(ParameterInfo parameter, JsonObject properties, JsonArray required)
+    {
+        var toolPropertyAttr = parameter.GetCustomAttribute<McpToolPropertyAttribute>();
+        if (toolPropertyAttr is null)
+        {
+            return false;
+        }
+
+        var propType = parameter.ParameterType.MapToToolPropertyType();
+        properties[toolPropertyAttr.PropertyName] = CreatePropertySchema(
+            propType.TypeName,
+            toolPropertyAttr.Description ?? string.Empty,
+            propType.IsArray,
+            propType.EnumValues);
+
+        if (toolPropertyAttr.IsRequired)
+        {
+            required.Add(toolPropertyAttr.PropertyName);
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Processes POCO parameters (generates schema from all public properties).
+    /// </summary>
+    private static bool TryProcessPocoParameter(ParameterInfo parameter, JsonObject properties, JsonArray required)
+    {
+        if (!parameter.ParameterType.IsPoco())
+        {
+            return false;
+        }
+
+        // POCOs should only be registered under McpToolTriggerAttribute
+        var toolTriggerAttribute = parameter.GetCustomAttribute<McpToolTriggerAttribute>();
+        if (toolTriggerAttribute is null)
+        {
+            return false;
+        }
+
+        GeneratePropertiesFromPoco(parameter.ParameterType, properties, required);
+        return true;
+    }
+
+    /// <summary>
+    /// Generates properties and required list from a POCO type directly into the provided collections.
+    /// </summary>
+    /// <param name="pocoType">The POCO type to analyze.</param>
+    /// <param name="properties">The JsonObject to populate with properties.</param>
+    /// <param name="required">The JsonArray to populate with required property names.</param>
+    private static void GeneratePropertiesFromPoco(Type pocoType, JsonObject properties, JsonArray required)
+    {
+        foreach (var property in pocoType.GetProperties(BindingFlags.Instance | BindingFlags.Public))
```

Comment:
We have a IsPoco helper class that you can use here to validate the type is actually a poco